### ISSUE TITLE
unable to catch 203 status code in dio try and catch block

### ISSUE THREAD

**Issue by keshavsharma5324** [flutterchina/dio#1295]
> ### New Issue Checklist
  try {
      if (params != null) {
        // Post Api
        final response = await _dio.post(api, data: params, options: options);
        return response.data;
      } else if (isDelete != null && isDelete == true) {
        // Delete Api
        final response = await _dio.delete(api);
        return response.data;
      } else {
        // Get Api
        final response = await _dio.get(api);
        return response.data;
      }
    } on SocketException {
      throw Exception('No Internet connection');
    } on DioError catch (e) {
      var msg = "";
      bool isUserExist = false;
      int statusCode = 0;
      var data;
      switch (e.type) {
        case DioErrorType.cancel:
          msg = 'Request Cancelled';
          statusCode = 0;
          break;
        case DioErrorType.connectTimeout:
          msg = 'Connection Timeout';
          statusCode = 0;
          // showMyDialog(
          //     title: 'Network Error!',
          //     msg: 'Failed to connect server. Connection timeout.',
          //     actions: <Widget>[
          //       TextButton(
          //           onPressed: () {
          //             NavigationService()
          //                 .navigationKey.currentState.pop();
          //           },
          //           child: Text(
          //             'Close',
          //             style: TextStyle(fontSize: 18),
          //           ))
          //     ]);
          break;
        case DioErrorType.receiveTimeout:
          msg = 'Connection Timeout';
          statusCode = 0;
          // showMyDialog(
          //     title: 'Network Error!',
          //     msg: 'Server taking to much time to send data.',
          //     actions: <Widget>[
          //       TextButton(
          //           onPressed: () {
          //             NavigationService()
          //                 .navigationKey.currentState.pop();
          //           },
          //           child: Text(
          //             'Close',
          //             style: TextStyle(fontSize: 18),
          //           ))
          //     ]);
          break;
        case DioErrorType.sendTimeout:
          msg = 'Connection Timeout';
          statusCode = 0;
          break;
        case DioErrorType.other:
          msg = 'No Internet Connection';
          statusCode = 0;
          break;
        case DioErrorType.response:
          switch (e.response!.statusCode) {
            case 203:
              msg ='invalid reponse of unauthorized request';
              statusCode = 203;
              break;
            case 202:
              msg ='invalid reponse of unauthorized request';
              statusCode = 202;
              break;
            case 302:
            // Bearer Token Missing
              msg = 'Invalid Request';
              statusCode = 302;
              break;
            case 400:
            // Bad Request
              msg = e.response!.data['message'];
              print("400 Bad Request => ${(e.response!.data as Map<String, dynamic>).containsKey("meta") ? e.response!.data["meta"]["is_user_exist"] : false}");
              isUserExist = (e.response!.data as Map<String, dynamic>).containsKey("meta") ? e.response!.data["meta"]["is_user_exist"] : false;
              statusCode = 400;
              data = e.response!.data['data'];
              break;
            case 401:
            // Un Authorized
              msg = e.response!.data['message'];
              statusCode = 401;

              break;
            case 403:
            // Forbidden
              msg = 'Forbidden Error';
              statusCode = 403;
              break;
            case 404:
            // Not Found
              msg = 'Api Not Found';
              statusCode = 404;
              break;
            case 500:
            // Internal Server Error
              msg = 'Server Error';
              statusCode = 500;
              break;
            case 503:
            // Service Unavailable
              msg = 'Service Unavailable';
              statusCode = 503;
              break;
            default:
              msg = 'Something went wrong';
              statusCode = 600;
              break;
          }
          break;
        default:
          msg = 'Something went wrong';
          statusCode = 601;
          break;
      }
      var dataToString = {
        "message": msg,
        "statusCode": statusCode,
        "is_user_exist": isUserExist,
        "data" : data
      };
      throw Exception(dataToString);
    } catch (e) {
      throw Exception('${e.toString()}');
    }
  }- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by markfili** [comment]
might be related to https://github.com/flutterchina/dio/issues/1306
